src/System.Memory/src/System/Buffers/ReadOnlyBufferExtensions.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#pragma warning disable 1591
+
+namespace System.Buffers
+{
+    public static class ReadOnlyBufferExtensions
+    {
+        public static SequencePosition? PositionOf<T>(this ReadOnlyBuffer<T> buffer, T value) where T: IEquatable<T>
+        {
+            SequencePosition position = buffer.Start;
+            SequencePosition result = position;
+            while (buffer.TryGet(ref position, out var memory))
+            {
+                var index = memory.Span.IndexOf(value);
+                if (index != -1)
+                {
+                    return buffer.GetPosition(result, index);
+                }
+                result = position;
+            }
+            return null;
+        }
+
+
+        /// <summary>
+        /// Copy the <see cref="ReadOnlyBuffer{T}"/> to the specified <see cref="Span{Byte}"/>.
+        /// </summary>
+        /// <param name="buffer">The source <see cref="ReadOnlyBuffer{T}"/>.</param>
+        /// <param name="destination">The destination <see cref="Span{Byte}"/>.</param>
+        public static void CopyTo<T>(this ReadOnlyBuffer<T> buffer, Span<T> destination)
+        {
+            if (buffer.Length > destination.Length)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.destination);
+            }
+
+            foreach (var segment in buffer)
+            {
+                segment.Span.CopyTo(destination);
+                destination = destination.Slice(segment.Length);
```

Comment:
I wonder how much faster it would be if we keep an integer index that we increment and use to Slice rather than re-assigning destination at each iteration. Maybe it is insignificant here.